lib/Root.php

```diff
@@ -292,66 +299,62 @@ private function coalesceIndexPath(string $dirPath): \Generator {
         }
     }
 
-    private function respond($fileInfo, Request $request, Response $response) {
+    private function respondFromFileInfo($fileInfo, Request $request): \Generator {
         // If the file doesn't exist don't bother to do anything else so the
         // HTTP server can send a 404 and/or allow handlers further down the chain
         // a chance to respond.
         if (empty($fileInfo->exists)) {
-            return;
+            $status = HttpStatus::NOT_FOUND;
+            return new Response\HtmlResponse(makeGenericBody($status), [], $status);
```

Comment:
Using `HtmlResponse` here defeats the Server's default [content type and charset options](https://github.com/amphp/aerys/blob/0.8.x/lib/Options.php#L18-L19)

Reply:
That was being defeated in several areas. This is an issue we should probably look at, perhaps with an error response handler.

Reply:
A server default shouldn't be a thing. Changing the server default breaks everything assuming that default otherwise.